### PR TITLE
fix(rtthread): implement lv_strcat function in rt-thread due to absence of rt_strcat

### PR DESCRIPTION
**Pull request by lhdjply**
### Description of the feature or fix

fix(warning): fix the issue of lv_string_rtthread prompting for the lack of strcat

image

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by FASTSHIFT — DISMISSED** [review]

File: src/stdlib/rtthread/lv_string_rtthread.c [hunk 1]
```diff
@@ -87,7 +87,14 @@ char * lv_strdup(const char * src)
...
 char * lv_strcat(char * dst, const char * src)
 {
-    return strcat(dst, src);
+    /*Since RT-thread does not have rt_strcat,
+    the following code is used instead.*/
+    char * tmp = dst;
+    while(*dst != '\0') {
+        dst++;
+    }
+    lv_strcpy(dst, src);
```
Comment (XuNeo) [inline src/stdlib/rtthread/lv_string_rtthread.c]:
```suggestion
    lv_strcpy(dst + lv_strlen(dst), src);
```

**Comment by kisvegabor** [comment]
Based on [this issue](https://github.com/lvgl/lvgl/issues/5897) we should have `lv_strncat`. @liamHowatt will update LVGL to use `lv_strncat`. I suggest updating this PR after that.

**Comment by lhdjply** [comment]
> Based on https://github.com/lvgl/lvgl/issues/5897 we should have lv_strncat. @liamHowatt will update LVGL to use lv_strncat. I suggest updating this PR after that.

Should I turn off my pr? Because I thought that when @liamHowatt was updated, the problem I solved with this pr should have been fixed. Or do I add the lv_strncat function directly to this pr? @kisvegabor

File: src/stdlib/rtthread/lv_string_rtthread.c [hunk 2]
```diff
@@ -87,7 +87,14 @@ char * lv_strdup(const char * src)
 
 char * lv_strcat(char * dst, const char * src)
 {
-    return strcat(dst, src);
+    /*Since RT-thread does not have rt_strcat,
+    the following code is used instead.*/
+    char * tmp = dst;
+    while(*dst != '\0') {
+        dst++;
+    }
```
Comment (XuNeo) [inline src/stdlib/rtthread/lv_string_rtthread.c]: Remove.

**Review by XuNeo — COMMENTED** [review]
We will have both lv_strcat and lv_strncat right?

File: src/stdlib/rtthread/lv_string_rtthread.c (hunk 2 above)
Comment (lhdjply) [reply src/stdlib/rtthread/lv_string_rtthread.c]: done

**Comment by liamHowatt** [comment]
Here is the PR #5927

**Review by XuNeo — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Comment by kisvegabor** [comment]
Thank you all for fixing it!

**Pull request merged**